gym-service/src/main/java/org/example/services/TrainingService.java

```diff
@@ -20,14 +19,22 @@
 @Service
 public class TrainingService {
 
-    private static final Logger LOGGER = LogManager.getLogger(TrainingService.class);
     @Autowired
     private TrainingRepository trainingRepository;
 
+    private TrainingEventSender trainingEventSender;
```

Comment:
curious why you use setter injection. Usually all the dependencies would be final and class will be annotated with `@RequiredArgsConstructor` annotation.
Unmodifiable dependencies helps to ensure thread-safety